io/net/bonding.py

```diff
@@ -0,0 +1,297 @@
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import process
+
+
+class Bonding(Test):
+    '''
+    Channel bonding enables two or more network interfaces to act as one,
+    simultaneously increasing the bandwidth and providing redundancy.
+    '''
+    def setUp(self):
+        '''
+        To check and install dependencies for the test
+        '''
+        sm = SoftwareManager()
+        depends = ["openssh-clients", "iputils"]
+        for pkg in depends:
+            if not sm.check_installed(pkg) and not sm.install(pkg):
+                self.skip("%s package is need to test" % pkg)
+        interfaces = netifaces.interfaces()
+        self.host_if1 = self.params.get("Iface1", default="")
+        self.host_if2 = self.params.get("Iface2", default="")
+        self.peer_if2 = self.params.get("peerif2", default="")
+        if self.host_if1 not in interfaces or self.host_if2 not in interfaces:
+            self.skip("interface is not available")
+        self.peer_ip = self.params.get("peerip", default="")
+        if self.peer_ip == "" or self.peer_if2 == "":
+            self.skip("peer machine should available")
+        msg = "ip addr show  | grep %s | grep -oE '[^ ]+$'" % self.peer_ip
+        cmd = "ssh %s %s" % (self.peer_ip, msg)
+        self.perr_if1 = process.system_output(cmd, shell=True).strip()
+        if self.perr_if1 == "":
+            self.fail("test failed because peer interface can not retrieved")
+        self.bond_name = self.params.get("bondname", default="tempbond")
+        self.mode = self.params.get("bonding_mode", default="").split(" ")
+        if self.mode == "":
+            self.skip("test skipped because mode not specified")
+        if self.host_if1 != "" and self.host_if2 != "" and self.peer_if2 != "":
+            cmd = "ip -f inet -o addr show %s | awk '{print $4}' | cut -d /\
+                  -f1" % self.host_if1
+            self.local_ip = process.system_output(cmd, shell=True).strip()
+            if self.local_ip == "":
+                self.fail("test failed because local ip can not retrieved")
+            cmd = "ip -f inet -o addr show %s | awk '{print $4}' | cut -d /\
+                  -f1" % self.host_if2
+            self.local_ip2 = process.system_output(cmd, shell=True).strip()
+            if self.local_ip2 == "":
+                self.fail("test failed because local ip can not retrieved")
+
+    def bond_remove(self, arg1):
+        '''
+        bond_remove
+        '''
+        if arg1 == "local":
+            cmd = "ifconfig %s down" % self.host_if1
+            if process.system(cmd, shell=True, ignore_status=True) != 0:
+                self.log.info("unable to bring down the interface")
+            cmd = "ifconfig %s down" % self.host_if2
+            if process.system(cmd, shell=True, ignore_status=True) != 0:
+                self.log.info("unable to bring down the interface")
+            cmd = "echo -%s > /sys/class/net/%s/bonding/slaves"\
```

Comment:
have
`bonding_slave_file = "/sys/class/net/%s/bonding/slaves"`

and access the interface specific file as:
`bonding_slave_file % host_iface1`